invokeai/frontend/web/src/features/stylePresets/hooks/usePresetModifiedPrompts.ts

```diff
@@ -0,0 +1,27 @@
+import { useAppSelector } from 'app/store/storeHooks';
+
+export const PRESET_PLACEHOLDER = `{prompt}`;
+
+export const buildPresetModifiedPrompt = (presetPrompt: string, currentPrompt: string) => {
+  return presetPrompt.includes(PRESET_PLACEHOLDER)
+    ? presetPrompt.replace(new RegExp(PRESET_PLACEHOLDER), currentPrompt)
+    : `${currentPrompt} ${presetPrompt}`;
+};
```

Comment:
The regex handles if the placeholder is present multiple times in the template - it should only replace the first one and this is the fix for that.

Reply:
Maybe I'm misunderstanding but this seems to work just fine:

```js
const PRESET_PLACEHOLDER = `[prompt]`;
let preset = `template template template ${PRESET_PLACEHOLDER} template template template ${PRESET_PLACEHOLDER}`;
preset.replace(PRESET_PLACEHOLDER, 'MY PROMPT');
// "template template template MY PROMPT template template template [prompt]"
preset = `template template template ${PRESET_PLACEHOLDER} template template template ${PRESET_PLACEHOLDER}`;
preset.replaceAll(PRESET_PLACEHOLDER, 'MY PROMPT');
// "template template template MY PROMPT template template template MY PROMPT" 
```

And with the prompt placeholder being `[prompt]`, it is actually valid regex, so the regex does this:
```js
 const PRESET_PLACEHOLDER = `[prompt]`;
let preset = `template template template ${PRESET_PLACEHOLDER} template template template ${PRESET_PLACEHOLDER}`;
preset.replace(new RegExp(PRESET_PLACEHOLDER), 'MY PROMPT');
// "MY PROMPTemplate template template [prompt] template template template [prompt]" 
```

Best to avoid regex unless really necessary. In this case we'd need to properly escape the regex first. 

BTW, `{` and `}` are regex syntax too - they quantify the preceding token. For example, `a{2}` matches `aa` but not `a`.